pkg/agent/monitortool/monitor.go

```diff
@@ -40,7 +40,7 @@ import (
 )
 
 // #nosec G404: random number generator not used for security purposes.
-var icmpEchoID = rand.Int31n(1 << 16)
+var icmpEchoID = rand.IntN(1 << 16)
```

Comment:
After this change, you should remove the explicit casts to `int` in the rest of the code. Look for occurrences of `int(icmpEchoID)` in this file.